### PR TITLE
Third time's the charm

### LINKED ISSUE #3673: 1.7.0-Beta: NoSuchMethodError: void kotlinx.coroutines.test.TestBuildersKt.runTest$default(kotlinx.coroutines.test.TestScope, long, kotlin.jvm.functions.Function2, int, java.lang.Object)
**Describe the bug**

When updating a project to use `kotlinx.coroutines` `1.7.0-Beta`, existing tests fail with a `NoSuchMethodError` for a `runTest` that's being called internally by the Compose test utilities.

The full stack trace looks like the following (but fail for every Compose test):

```
com.alexvanyo.composelife.resourcestate.ResourceStateComposableTests > collect_as_state_is_correct FAILED
    java.lang.NoSuchMethodError: 'void kotlinx.coroutines.test.TestBuildersKt.runTest$default(kotlinx.coroutines.test.TestScope, long, kotlin.jvm.functions.Function2, int, java.lang.Object)'
        at androidx.compose.ui.test.AndroidComposeUiTestEnvironment.withTestCoroutines(ComposeUiTest.android.kt:342)
        at androidx.compose.ui.test.AndroidComposeUiTestEnvironment.access$withTestCoroutines(ComposeUiTest.android.kt:218)
        at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1$1.invoke(ComposeUiTest.android.kt:294)
        at androidx.compose.ui.test.junit4.IdlingStrategy.withStrategy(IdlingStrategy.android.kt:52)
        at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1$1.invoke(ComposeUiTest.android.kt:293)
        at androidx.compose.ui.test.junit4.IdlingResourceRegistry.withRegistry(IdlingResourceRegistry.jvm.kt:157)
        at androidx.compose.ui.test.AndroidComposeUiTestEnvironment$runTest$1.invoke(ComposeUiTest.android.kt:292)
        at androidx.compose.ui.test.junit4.ComposeRootRegistry.withRegistry(ComposeRootRegistry.android.kt:146)
        at androidx.compose.ui.test.AndroidComposeUiTestEnvironment.runTest(ComposeUiTest.android.kt:291)
        at androidx.compose.ui.test.ComposeUiTest_androidKt.runAndroidComposeUiTest(ComposeUiTest.android.kt:107)
        at androidx.compose.ui.test.ComposeUiTest_androidKt.runComposeUiTest(ComposeUiTest.android.kt:63)
        at com.alexvanyo.composelife.resourcestate.ResourceStateComposableTests.collect_as_state_is_correct(ResourceStateComposableTests.kt:39)
```

**Provide a Reproducer**

Reproducing PR: https://github.com/alexvanyo/composelife/pull/838

It looks like the update to `runTest` in `1.7.0-Beta` might not have been binary compatible?

### PR DESCRIPTION
**Pull request by dkhalanskyjb**
Fixes #3673 
Thanks, @ephemient!

Context: https://github.com/Kotlin/kotlinx.coroutines/pull/3742#discussion_r1192265210

### CODE REVIEW

**Comment by dkhalanskyjb** [comment]
Actually, this isn't *required*. If we look at the sources of how this is used in the problematic library, the dispatch timeout is not used at all, `runTest` is only called as a workaround when no coroutines are even left to run. Still, it's better to be safe than sorry.

**Review by qwwdfsad — APPROVED** [review]
Please add a rationale and short explanation to the squashed commit message

File: kotlinx-coroutines-test/common/src/TestBuilders.kt [hunk 1]
```diff
@@ -580,6 +580,6 @@ internal expect fun dumpCoroutines()
 public fun TestScope.runTestLegacy(
     dispatchTimeoutMs: Long,
     testBody: suspend TestScope.() -> Unit,
-    unused1: Int,
+    marker: Int,
     unused2: Any?,
```
Comment (ephemient) [inline kotlinx-coroutines-test/common/src/TestBuilders.kt:584]: Maybe rename the parameters to `mask: Int, marker: Any?`.

Comment (ephemient) [reply kotlinx-coroutines-test/common/src/TestBuilders.kt:584]: Ah, I'm too slow. Well it shouldn't really matter.

Comment (dkhalanskyjb) [reply kotlinx-coroutines-test/common/src/TestBuilders.kt:584]: I disagree. `unused2` gives this function a tongue-in-a-cheek charm, which it deserves.

**Pull request merged**